### PR TITLE
Add community/protobuf-c plugin

### LINKED ISSUE #894: Plugin request for Buf Schema Registry - protobuf-c/protobuf-c
## Mandatory

**Where is the source code for the plugin?**

https://github.com/protobuf-c/protobuf-c

## Optional

**Does the plugin have a valid semver version?**

Yes, 1.4.1

**Does the plugin have a dependency on another plugin?**

No

**Does the output of this plugin depend on any other external libraries?**

No

### PR DESCRIPTION
**Pull request by pkwarren**
Fixes #894.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.21
 
 require (
 	aead.dev/minisign v0.2.1
-	github.com/bufbuild/buf v1.28.1
+	github.com/bufbuild/buf v1.28.2-0.20231128013053-57a9b07c7833
```
Comment (pkwarren) [inline go.mod:7]: Needed for https://github.com/bufbuild/buf/pull/2626

File: .github/workflows/arm64.yml [hunk 2]
```diff
@@ -38,9 +38,8 @@ jobs:
         go-version: '1.21'
         check-latest: true
     - name: Install buf cli
-      uses: bufbuild/buf-setup-action@v1
-      with:
-        github_token: ${{ github.token }}
+      run: |
+        go install github.com/bufbuild/buf/cmd/buf
```
Comment (mfridman) [inline .github/workflows/arm64.yml:42]: Don't you need the `@version` ? e.g., `@latest`

Comment (pkwarren) [reply .github/workflows/arm64.yml:42]: I think it should use the version from the go.mod file? 🤞

Comment (mfridman) [reply .github/workflows/arm64.yml:42]: Ye, momentary lapse of judgment.

Comment (pkwarren) [reply .github/workflows/arm64.yml]: Didn't work - added version.

File: plugins/community/protobuf-c/v1.5.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN apt-get update \
+ && apt-get install -y build-essential curl libprotobuf-dev libprotoc-dev pkg-config protobuf-compiler
+
+WORKDIR /build
+RUN curl -fsSL -o /tmp/protobuf-c.tar.gz https://github.com/protobuf-c/protobuf-c/releases/download/v1.5.0/protobuf-c-1.5.0.tar.gz \
+ && tar zxf /tmp/protobuf-c.tar.gz \
+ && cd protobuf-c-1.5.0 \
+ && ./configure \
+ && make LDFLAGS=-all-static -j$(nproc) \
```
[the remark this answers is not in the document]
Comment (mfridman) [reply plugins/community/protobuf-c/v1.5.0/Dockerfile:13]: awesome work!

**Review by mfridman — APPROVED** [review]

**Pull request merged**